cache/remotecache/s3/s3.go

```diff
@@ -408,6 +408,9 @@ func (s3Client *s3Client) getManifest(ctx context.Context, key string, config *v
 	if err := decoder.Decode(config); err != nil {
 		return false, errors.WithStack(err)
 	}
+	if _, err := decoder.Token(); !errors.Is(err, io.EOF) {
```

Comment:
Curious; I recall we used `decoder.More()` to detect if there's additional content elsewhere. Would that work as well, or would that only look for valid JSON objects and ignore invalid content? https://github.com/moby/moby/blob/f3d377e4221903183faee08888da7d4bacf74450/api/server/httputils/httputils.go#L83

Reply:
`More` doesn't seem to work correctly for some input characters https://go.dev/play/p/X4tmG7orx9W

Reply:
Ah! That's what I somewhat started to suspect when I saw the GoDoc mention "if there's another _element_". I wasn't sure if that would include _invalid_ elements; thanks!

Looks like we need to update that code (and perhaps contribute a GoDoc improvement in Golang)